### DISCUSSION TITLE
Mini-Batch ADB requests to speed up execution

### DISCUSSION THREAD

**Discussion by TurpIF** [Banno/Gordon#49] [Ideas]
Hello,

I'm trying this plugin and I noticed a really big slowness in the execution of Android tests: going from ~25 minutes to ~50 minutes (and I'm counting emulator startup+gradle compilation whose the duration is fixed).
I saw that there is one ADB command per test methods. But ADB command may be slow and maybe it would be possible to let the user configure a size of batch so that many test methods could be executed at the same time.

**Comment by joshschriever** [comment]
I'll leave this open as a possibility to look into, but what I would really recommend is using multiple devices and using the `SinglePool` or `PhonesAndTablets` pooling strategy - that's likely the best way to save time.
